src/main/java/org/elasticsearch/common/lucene/search/function/ScoreFunction.java

```diff
@@ -19,18 +19,16 @@
 
 package org.elasticsearch.common.lucene.search.function;
 
-import org.apache.lucene.index.LeafReaderContext;
 import org.apache.lucene.search.Explanation;
+import org.elasticsearch.common.lucene.ReaderContextAware;
 
 /**
  *
  */
-public abstract class ScoreFunction {
+public abstract class ScoreFunction implements ReaderContextAware {
```

Comment:
@jpountz For my own understanding: ScoreFunction is unrelated to Aggregations right?